File: language-support/js/daml-ledger/tests/arbitrary/get_transaction_by_id_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as jsc from 'jsverify';
import * as ledger from '../../src';

export const GetTransactionByIdRequest: jsc.Arbitrary<ledger.GetTransactionByIdRequest> =
    jsc.tuple([jsc.string, jsc.array(jsc.string)]).smap(
        ([transactionId, requestingParties]) => ({
            transactionId: transactionId,
            requestingParties: requestingParties
        }),
        (request) =>
            [request.transactionId, request.requestingParties]
    );

File: sdk/canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/ledger/api/messages/transaction/GetTransactionByIdRequest.scala
// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.canton.ledger.api.messages.transaction

import com.digitalasset.canton.ledger.api.domain.UpdateId
import com.digitalasset.daml.lf.data.Ref.Party

final case class GetTransactionByIdRequest(
    updateId: UpdateId,
    requestingParties: Set[Party],
)
